sharding/contracts/validator_manager.v.py

```diff
@@ -286,176 +285,61 @@ def get_member_of_committee(
-    # Check if only one collation in one period perd shard
-    assert self.period_head[shard_id] < expected_period_number
+        chunk_root: bytes32
+    ) -> bool:
+
+    # Check that shard_id is valid
+    assert shard_id >= 0 and shard_id < self.SHARD_COUNT
+    # Check that it's current period
+    current_period: int128 = floor(block.number / self.PERIOD_LENGTH)
+    assert current_period == period
+    # Check that no header is added yet in this period in this shard
+    assert self.records_updated_period[shard_id] < period
 
     # Update notary_sample_size
     self.update_notary_sample_size()
 
-    # Check if this header already exists
-    header_bytes: bytes <= 288 = concat(
-        convert(shard_id, 'bytes32'),
-        convert(expected_period_number, 'bytes32'),
-        period_start_prevhash,
-        parent_hash,
-        transaction_root,
-        convert(collation_coinbase, 'bytes32'),
-        state_root,
-        receipt_root,
-        convert(collation_number, 'bytes32'),
-    )
-    entire_header_hash: bytes32 = convert(
-        uint256_mod(
-            convert(sha3(header_bytes), 'uint256'),
-            # Mod 2^208, i.e., extract right most 26 bytes
-            convert(411376139330301510538742295639337626245683966408394965837152256, 'uint256')
-        ),
-        'bytes32'
-    )
-    
-    # Check if parent header exists.
-    # If it exist, check that it's score is greater than 0.
-    parent_collation_score: int128 = self.get_collation_header_score(
-        shard_id,
-        parent_hash,
-    )
-    if not not parent_hash:
-        assert parent_collation_score > 0
-
-    # Check score == collation_number
-    _score: int128 = parent_collation_score + 1
-    assert collation_number == _score
-
-    # Add the header
-    self.collation_headers[shard_id][entire_header_hash] = convert(
-        uint256_add(
-            uint256_mul(
-                convert(parent_hash, 'uint256'),
-                # Multiplied by 2^48, i.e., left shift 6 bytes
-                convert(281474976710656, 'uint256')
-            ),
-            uint256_mod(
-                convert(_score, 'uint256'),
-                # Mod 2^48, i.e. confine it's range to 6 bytes
-                convert(281474976710656, 'uint256')
-            ),
-        ),
-        'bytes32'
-    )
-
-    # Update the latest period number
-    self.period_head[shard_id] = expected_period_number
+    # Add header
+    self.collation_records[period][shard_id] = {
+        chunk_root: chunk_root,
+        proposer: msg.sender,
+        is_elected: False,
+    }
 
-    # Determine the head
-    is_new_head: bool = False
-    shard_head_score: int128 = self.get_collation_header_score(
-        shard_id,
-        self.shard_head[shard_id],
-    )
-    if _score > shard_head_score:
-        self.shard_head[shard_id] = entire_header_hash
-        is_new_head = True
+    # Update records_updated_period
+    self.records_updated_period[shard_id] = current_period
 
```

Comment:
since it's a new header, do we want to clear `current_vote[shard_id]`'s last byte to 0 here?

Reply:
Yes we do! I plan to do this along with `submit_vote` because then others can get a more clear picture about how `current_vote` is used when they look into the commit history.

Reply:
Got it. Thanks! I'm doing the same implementation in solidity. Feel free to take a look and leave me any feedback : )
https://github.com/prysmaticlabs/geth-sharding/pull/97